pcsx2/CDVD/CDVD.cpp

```diff
@@ -1416,6 +1417,10 @@ static __fi void cdvdWrite0F(u8 rt) { // TYPE
 static __fi void cdvdWrite14(u8 rt) { // PS1 MODE?? // This should be done in the SBUS_F240 bit 19 write in HwWrite.cpp
 	u32 cycle = psxRegs.cycle;
 
+	PSXCLK = 33868800;      //33.8688 Mhz Playstation 1 speed.
+    psxEEmultiplier = 8.71; //proper EE to iop cycle freaquency.
```

Comment:
What is the data type of psxEEmultiplier here? Integer, correct?
It is an external integer in the other files, so unless some C standard addition added some automatic promotion, this line says psxEEmultiplier = 8; ;p

Reply:
It's a float value to hold the ps1's 8.71

Reply:
Print out that value to console and see what it says.

Reply:
It trunacates to a 9

Reply:
Truncates is right but the value is 8. That's the default behaviour for when you fit a float into an integer.
Anyway, this part is obviously not working right, but it's still plausible that this pr fixes PSX games.
Must be the cycle correction then?

Reply:
Well, it must be either the PSXCLK variable or the iopCycleEE -= 9. 
Those are the effective changes compared to master.

It looks like psxEEmultiplier used to be a float before?
In that case, this pr definitely needs to be re-tested, to see if it still does the thing ;)

Reply:
yeah the divider should be 8.71 but there's no mechanism in there currently to account for the remainder to accurately (or semi accurately) control the cycles consumed. So it gets rounded to 9 which is still closer than it was when it was fixed to 8 xD

Reply:
It's just that the code as is right now has psxEEmultiplier as an integer. The assignment of "8.71" to this is the same as psxEEmultiplier = 8. But from the earlier discussion, it looks like psxEEmultiplier used to be a float, so I'm asking :p

Reply:
It was, but it doesn't make a difference on the performance of this pro the effect a float vs an int has in this case from my tests have been the same

Reply:
Yeah, I'm sure that changing timings will fix some games. I've seen this earlier with CDVD timing especially when I slowed that down (far too much, just as a test).
The task now is to make it correct, and at least the idea of this pr is the right direction.

We should start by figuring out what change in this pr is fixing the games.
Other than that, we already know that setting the value of PSXCLK dynamically is correct.
(And it doesn't require anything fancy.)

Reply:
The interesting thing is that dma3 can't actually handle that speed in ff viii there's dma3 related hung timeouts that occur exclusively to this build and metal gear solid which previously worked I iop's interpreter no longer starts flashing bad opcodes in the console

Reply:
That could be because of this

cdReadTime = ((PSXCLK / 75) / BIAS);

That's a fixed read time, we don't know if that's right, plus it can vary depending on seek times, if the disc is spinning etc.

Reply:
It is, they happen less if that's not set

Reply:
What happens less if it's not set?

Reply:
Hung timeouts in ff viii

Reply:
so it turns out it's a combo of PSXCLK cdreadspeed and iopCycleEE -= 9 that breaks metal gear solid but fixes kolona, medievil was fixed before my tampering with it.